include/ck/tensor_operation/gpu/grid/gridwise_softmax.hpp

```diff
@@ -303,9 +305,10 @@ struct GridwiseSoftmax_mk_to_mk
             reducedTiles++;
         } while(reducedTiles < num_k_block_tile_iteration);
 
+        block_sync_lds(); // wait for reading being complete before writing to LDS
         static_for<0, MThreadSliceSize, 1>{}([&](auto I) {
             BlockwiseSumReduce::Reduce(reduce_work_buf, accu_value_buf(I));
-            // block_sync_lds();
+            block_sync_lds();
```

Comment:
How about add follwing before BlockwiseSumReduce?
`if constexpr(I>0) 
  block_sync_lds()'
This may save one snyc

Reply:
I have a fix planned for another PR here https://github.com/ROCmSoftwarePlatform/composable_kernel/commit/4aa3754a6771e70ffa4f49dcf5cc49c4996aa1ac. Let's merge this PR to unblock develop branch asap